mirrord/agent/src/steal/connection.rs

```diff
@@ -147,10 +157,66 @@ impl Client {
         }
 
         let framed = HTTP_FRAMED_VERSION.matches(&self.protocol_version);
+        let chunked = HTTP_CHUNKED_VERSION.matches(&self.protocol_version);
         let tx = self.tx.clone();
 
         tokio::spawn(async move {
-            if framed {
+            // Chunked data is preferred over framed data
+            if chunked {
+                let connection_id = request.connection_id;
+                let request_id = request.request_id;
+                // If frames are immediately available, send them
+                let mut ready_frames = vec![];
+                let mut frame = request.request.frame();
+                loop {
+                    match poll!(&mut frame) {
+                        Poll::Ready(Some(ready_frame)) => {
+                            ready_frames.push(ready_frame.unwrap());
+                            frame = body.frame();
+                            continue;
+                        }
+                        _ => break,
+                    }
+                }
+                for frame in ready_frames {
+                    let Ok(request) = request.into_serializable_chunked(frame.into_data()).await
```

Comment:
The other variant of `Frame` is also interesting (`frame.into_trailers()`), we want to send it as well